Add first and last page buttons to transaction page

mustardpower opened the pull request:
Can now navigate to first and last page via two additional buttons. Need some limitation to prevent navigation past the last page?

Closes #3.

mustardpower committed:
Add first and last buttons to transaction page

Can now navigate to first and last page via two additional buttons. Might need some limitation to prevent navigation past the last page?

franky47 commented on src/client/components/PageNavigation.tsx:
```diff
@@ -2,7 +2,9 @@ import { Button, Flex, FlexProps, Text } from '@chakra-ui/react'
 import React from 'react'
 import { FiArrowLeft, FiArrowRight } from 'react-icons/fi'
+import { MdFirstPage, MdLastPage} from 'react-icons/md'
 import { useTransactionPage } from 'src/services/centralized-coin'
+import { getNumberOfPages } from 'src/services/centralized-coin'
```
Could you merge the imports please ?

franky47 commented on src/client/components/PageNavigation.tsx:
```diff
@@ -11,17 +13,26 @@ function useNavigation() {
   })
+  
+  const last = React.useCallback(() => getNumberOfPages().then(
+    (value: number) => { setPage(page => value)
```
This line breaks when building, probably just `setPage(value)` ?

franky47 commented on src/client/components/PageNavigation.tsx:
```diff
@@ -11,17 +13,26 @@ function useNavigation() {
     []
   )
+
+  const first = React.useCallback(() => setPage(page => 0), [])
```
Same here as above.

mustardpower committed:
Fix build errors relating to unused page parameters

No need to use lambda in this case, just pass the value.

franky47 reviewed (approved):
LGTM!

mustardpower commented:
There is actually an error with the number of pages calculation - doesn't currently account for any remaining transactions on the last page that isn't completely full.

franky47 commented on src/services/centralized-coin.ts:
```diff
@@ -104,3 +104,13 @@ export function usePreloadNeighbouringPages(pageNumber: number) {
+  const data = await res.json()
+  const transactionsPerPage = 50;
+  const numberOfTransactions: number = data.count
+  return Math.floor(numberOfTransactions/ transactionsPerPage);
```
This should probably be `Math.ceil`, to account for the last page.

Eg, if there were 55 transactions, 55/50 = 1.1, there would be 2 pages.

mustardpower commented:
I am unsure about the page number calculation. I checked there are 50 transactions on the page but the api endpoint is returning 2772 at this moment in time. However, in the transaction explorer I see only 2 transactions on the last page. Should be 22?

mustardpower committed:
Fix for page numbers calculation

Number of pages should be rounded up to make up for the last page which may be incomplete. Also renamed the function because the number of pages is actually the result of this function + 1 (because pages are zero indexed).

franky47 reviewed (approved):
LGTM! Do you want the coins on your Git committer email ? Otherwise, send me your address at contact@francoisbest.com
